src/crowbar.mli

```diff
@@ -1,52 +1,128 @@
-type state
-
-type 'a printer = Format.formatter -> 'a -> unit
+(** {1:top Types } *)
 
 type 'a gen
+(** ['a gen] knows how to generate ['a] for use in Crowbar tests. *)
 
 type ('k, 'res) gens =
   | [] : ('res, 'res) gens
   | (::) : 'a gen * ('k, 'res) gens -> ('a -> 'k, 'res) gens
+(** multiple generators are provided to functions using a listlike syntax.
+    for example, [map [int; int] (fun a b -> a + b)] *)
+
+type 'a printer = Format.formatter -> 'a -> unit
+(** pretty-printers for items generated by Crowbar; useful for the user in
+    translating test failures into bugfixes. *)
+
+(**/**)
 (* re-export stdlib's list
    We only want to override [] syntax in the argument to Map *)
 type nonrec +'a list = 'a list = [] | (::) of 'a * 'a list
+(**/**)
+
+(** {1:generators Generators } *)
+
+(** {2:simple_generators Simple Generators } *)
+
+val int : int gen
+(** [int] generates an integer.  For 32-bit systems, it will generate an
+    Int32.t; otherwise, it will generate an Int64.t .  If you need integers from
+    a smaller domain, consider using {!range}. *)
```

Comment:
It would be more accurate to say that it generates an integer ranging from min_int to max_int, inclusive, and avoid mentioning Int32/Int64. (Internally, yes they are used, but results are truncated to 31 or 63 bits by Int{32,64}.to_int)